engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb

```diff
@@ -19,8 +20,15 @@ def call!(redirect_url = full_order_path(@order))
           @order.errors.add(:base, I18n.t('authorization_failure'))
         end
 
-        @payment
+        return @payment unless @payment.requires_authorization? && off_session
+
+        PaymentMailer.authorize_payment(@payment).deliver_now
+        PaymentMailer.authorization_required(@payment).deliver_now
```

Comment:
It's really a cosmetic detail and maybe not the ruby way either, but as a reader I would have found it clearer with somthing like that:

```ruby
need_asynchronous_payment_authorization = @payment.requires_authorization? && off_session

if need_asynchronous_payment_authorization
    PaymentMailer.authorize_payment(@payment).deliver_now
    PaymentMailer.authorization_required(@payment).deliver_now
else
    return @payment
end

Reply:
Just saw afterwards that you also edited the code so my comment is not very useful.